internal/consumer/service.go

```diff
@@ -0,0 +1,81 @@
+package consumer
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/emmaLP/gs-software-onboarding/internal/model"
+
+	"github.com/emmaLP/gs-software-onboarding/pkg/hackernews"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+)
+
+type Service struct {
+	logger          *zap.Logger
+	numberOfWorkers int
+	hnClient        hackernews.Client
+}
+
+type Client interface {
+	processStories(ctx context.Context)
+}
+
+func NewService(logger *zap.Logger, config *model.ConsumerConfig, hnClient hackernews.Client) (*Service, error) {
+	if hnClient == nil {
+		var err error
+		hnClient, err = hackernews.New(config.BaseUrl, nil)
+		if err != nil {
+			return nil, fmt.Errorf("Unable to create HackerNew client: %w", err)
+		}
```

Comment:
You dont need to do this here,  you're passing `hnClient` as nil so this can just be the base logic and remove your `hnClient` param

Reply:
I did this to support testing so I could inject a mock hnclient over the actual client so that during unit tests I didn't call out to the hackernews api but instead mocked the calls